graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java

```diff
@@ -127,6 +131,8 @@ public StreamDestinationFilterRuleDTO createForStream(String streamId, StreamDes
             throw new IllegalArgumentException("id must be blank");
         }
 
+        optionalDestinationFilterCreationValidator.ifPresent(validator -> validator.validate(dto));
```

Comment:
Is there a chance that a client will move from enterprise to open and after that face dto with `data-warehouse` destination type here? If yes, isn't it a potential bug that this dto won't be validated?

Reply:
@kodjo-anipah -> I am leaving the decision to you, just wanted to know if you are aware of this scenario. :)